src/ddmd/backend/cgen.c

```diff
@@ -709,7 +709,10 @@ struct FixupArray
     {
         if (dim == cap)
         {
-            cap = cap ? 2 * cap : 16;
+            // 0x800 determined experimentally to minimize reallocations
+            cap = cap
+                ? (3 * cap) / 2 // use 'Tau' of 1.5
```

Comment:
I assume cap never decreases, so it's either 0 or something no less than 0x800. Otherwise the code has a bug if cap = 1.

Reply:
`cap` starts at 0, and is set to 0x800 upon first use. It is never `0<cap<0x800`.